src/containers/my-resources/lessons-container/LessonsContainer.tsx

```diff
@@ -0,0 +1,179 @@
+import { ChangeEvent, useCallback, useState } from 'react'
+import { useTranslation } from 'react-i18next'
+
+import SearchIcon from '@mui/icons-material/Search'
+import { Box } from '@mui/material'
+
+import { useNavigate } from 'react-router-dom'
+import AppButton from '~/components/app-button/AppButton'
+import AppDrawer from '~/components/app-drawer/AppDrawer'
+import AppPagination from '~/components/app-pagination/AppPagination'
+import EnhancedTable from '~/components/enhanced-table/EnhancedTable'
+import InputWithIcon from '~/components/input-with-icon/InputWithIcon'
+import Loader from '~/components/loader/Loader'
+import {
+  columns,
+  defaultResponse,
+  initialSort,
+  itemsLoadLimit,
+  removeColumnRules
+} from '~/containers/my-resources/lessons-container/LessonsContainer.constants'
+
+import usePagination from '~/hooks/table/use-pagination'
+import useSort from '~/hooks/table/use-sort'
+import useAxios from '~/hooks/use-axios'
+import useBreakpoints from '~/hooks/use-breakpoints'
+import useConfirm from '~/hooks/use-confirm'
+import { useDrawer } from '~/hooks/use-drawer'
+import { ResourceService } from '~/services/resource-service'
+import {
+  ajustColumns,
+  createUrlPath,
+  getScreenBasedLimit
+} from '~/utils/helper-functions'
+
+import { authRoutes } from '~/router/constants/authRoutes'
+
+import { styles } from '~/containers/my-resources/lessons-container/LessonsContainer.styles'
+
+import { useDebounce } from '~/hooks/use-debounce'
+import { ItemsWithCount, Lesson } from '~/types'
+
+const LessonsContainer = () => {
+  const breakpoints = useBreakpoints()
+  const { t } = useTranslation()
+  const navigate = useNavigate()
+
+  const { openDialog } = useConfirm()
+  const { openDrawer, closeDrawer, isOpen } = useDrawer()
+
+  const [searchTitle, setSearchTitle] = useState('')
+  const [searchInput, setSearchInput] = useState('')
```

Comment:
Do you really need these two states? Because you can pass searchTitle as a value to input

Reply:
yes, cause as you see in the code it won't work with debounce properly